src/Microsoft.Health.Client/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,92 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Net.Http;
+using EnsureThat;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Client.Configuration;
+using Microsoft.Health.Extensions.DependencyInjection;
+using IHttpClientFactory = Microsoft.IdentityModel.Clients.ActiveDirectory.IHttpClientFactory;
+
+namespace Microsoft.Health.Client.Extensions;
+
+public static class ServiceCollectionExtensions
+{
+    public static void AddNamedManagedIdentityCredentialProvider(this IServiceCollection serviceCollection, IConfiguration managedIdentityCredentialConfiguration, string name)
+    {
+        EnsureArg.IsNotNull(serviceCollection, nameof(serviceCollection));
+        EnsureArg.IsNotNull(managedIdentityCredentialConfiguration, nameof(managedIdentityCredentialConfiguration));
+        EnsureArg.IsNotNullOrWhiteSpace(name, nameof(name));
+
+        serviceCollection.Configure<ManagedIdentityCredentialConfiguration>(name, managedIdentityCredentialConfiguration);
+
+        serviceCollection.Add(provider =>
+            {
+                IOptionsMonitor<ManagedIdentityCredentialConfiguration> options = provider.GetService<IOptionsMonitor<ManagedIdentityCredentialConfiguration>>();
+                var httpClientFactory = provider.GetService<IHttpClientFactory>();
```

Comment:
Why is this credential provider the only one that uses HttpClientFactory?

Reply:
The `ManagedIdentityCredentialProvider` uses a `AzureServiceTokenProvider` under the hood and it's constructor requires an `IHttpClientFactory`. It looks like we can actually switch this to a `DefaultAzureCredential` and remove this need.